controllers/control_plane_init.go

```diff
@@ -0,0 +1,48 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/cloudinit"
+)
+
+type UserDataGenerator struct {
+	logr.Logger
+}
+
+func (u *UserDataGenerator) GenerateControlPlaneInit(config *v1alpha1.KubeadmBootstrapConfig) error {
+	u.Info("Generating user data for first control plane")
+
+	// TODO: sub in things in the cluster configuration like the cluster name and other data we can extract from the cluster object
+	config.Spec.ClusterConfiguration.APIVersion = "v1beta1"
```

Comment:
What if the users pass a wrong value in this field? are we overriding silently or should we fail/warn

Reply:
In the past we have issued a message when a user tried to provide default value that did not match what our code enforces.

I believe we should do the same and issue a log